x-pack/plugins/ml/public/cases/anomaly_charts_attachments.tsx

```diff
@@ -5,20 +5,95 @@
 import { FormattedMessage } from '@kbn/i18n-react';
 import type { PersistableStateAttachmentViewProps } from '@kbn/cases-plugin/public/client/attachment_framework/types';
 import { FIELD_FORMAT_IDS } from '@kbn/field-formats-plugin/common';
-import { EuiDescriptionList } from '@elastic/eui';
+import { EuiDescriptionList, htmlIdGenerator } from '@elastic/eui';
 import type { FieldFormatsStart } from '@kbn/field-formats-plugin/public';
-import type { AnomalyChartsEmbeddableInput } from '../embeddables';
+import type { TimeRange } from '@kbn/es-query';
+import { isPopulatedObject } from '@kbn/ml-is-populated-object';
+import { LazyEmbeddableAnomalyChartsContainer } from '../embeddables/anomaly_charts/lazy_anomaly_charts_container';
+import { initializeAnomalyChartsControls } from '../embeddables/anomaly_charts/initialize_anomaly_charts_controls';
+import type {
+  AnomalyChartsApi,
+  AnomalyChartsEmbeddableCustomInput,
+  AnomalyChartsEmbeddableState,
+  AnomalyChartsEmbeddableServices,
+} from '../embeddables';
+
+interface AnomalyChartsCaseAttachmentProps extends AnomalyChartsEmbeddableCustomInput {
+  services: AnomalyChartsEmbeddableServices;
+}
+const AnomalyChartsCaseAttachment = ({
+  services,
+  ...initialState
+}: AnomalyChartsCaseAttachmentProps) => {
+  const id = useMemo(() => htmlIdGenerator()(), []);
+  const [coreStartServices, pluginStartServices, mlServices] = services;
+  const contextServices = useMemo(
+    () => ({
+      mlServices: {
+        ...mlServices,
+      },
+      ...pluginStartServices,
+      ...coreStartServices,
+    }),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    []
+  );
+
+  const api = useMemo(
+    () => {
+      const anomalyChartsApi = initializeAnomalyChartsControls(initialState);
+      const combined: AnomalyChartsApi = {
+        ...anomalyChartsApi.anomalyChartsControlsApi,
+        ...anomalyChartsApi.dataLoadingApi,
+      };
+      return combined;
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    []
+  );
 
-export const initComponent = memoize(
-  (fieldFormats: FieldFormatsStart, EmbeddableComponent: FC<AnomalyChartsEmbeddableInput>) => {
+  const timeRange$ = useMemo(
+    () => new BehaviorSubject(initialState.timeRange),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [initialState.timeRange?.from, initialState.timeRange?.to]
+  );
+
+  return (
+    <div css={css({ display: 'flex', width: '100%' })}>
+      <KibanaRenderContextProvider {...coreStartServices}>
+        <KibanaContextProvider services={contextServices}>
+          <LazyEmbeddableAnomalyChartsContainer
+            id={`case-anomaly-charts-${id}`}
+            severityThreshold={initialState.severityThreshold}
+            api={api}
+            services={services}
+            onLoading={api.onLoading}
+            onRenderComplete={api.onRenderComplete}
+            onError={api.onError}
+            timeRange$={timeRange$}
```

Comment:
This is a misnomer that was leftover. The case attachment is actually not consuming an embeddable - it's just passing in the data so that it can be rendered. I've renamed it [here](https://github.com/elastic/kibana/pull/183456/commits/84136d7ea95c1a3a3c4395a64e636e1a44bc92a1)